rfcs/text/0000-matrix-service.md

```diff
@@ -0,0 +1,418 @@
+
+#### Response
+
+The response object returns a `request_id` that users can use to retrieve the status and results of a request by supplying it to the `GET` `/matrix` endpoint. A `status` and `message` field are also supplied to inform the user whether the request started successfully or not.
+
+*Note about multiple species:*
+If a request selects cells from multiple species, the Matrix Service will generate one matrix per species and thus return multiple `request_id`s. The convention here is for the human matrix to be reflected in `request_id` and all other species matrices to be reflected in a list of `request_id`s via `non_human_request_ids`.
+
+```
+{
+	'request_id': string,
+	'non_human_request_ids': list,
+	'status': one of ["In Progress", "Complete", "Failed", "Expired"],
+	'message': string
+}
+```
+
+### GET /matrix/<request_id>
+
+A `request_id` returned by the `POST` endpoint can be supplied as a URL parameter to this endpoint to retrieve the status and results of the request.
+
+#### Response
+
+```
+{
+	'request_id': string,
+	'status': one of ["In Progress", "Complete", "Failed", "Expired"],
+	'matrix_url': string,
+	'message': string
+}
+```
+
+The response returns the `request_id`, a `status` and `message` similar to the response of the `POST /matrix` endpoint. If the `status` of the request is `Complete`, then `matrix_url` will be populated with the public S3 download URL to the generated matrix. Otherwise, an empty string will be returned and a message explaining the status or an error that interrupted the request. For information on expired requests, see [Availability](#availability).
+
+### Auxiliary Endpoints
+
+To improve user experience, for each of the four POST body parameters, a pair of `GET` endpoints are available to provide more information about the parameter and the set of accepted values. For example, `GET /filters` and `GET /filters/<filter>` provide respectively, the full list of acceptable filters and information about a specific filter. For filters that take on categorical (string) values, the latter will return a description of the filter, the set of valid values for the filter, and a distribution of cell counts across the values. Numeric filters, such as `genes_detected`, will return a description and the maximum and minimum values across all cells . An example response from each endpoint illustrates this:
+
+Ex. GET /filters
+
+```
+[
+    "cell_suspension.provenance.document_id",
+    "genes_detected",
+    "total_umis",
+    "barcode",
+    "emptydrops_is_cell",
+    "specimen_from_organism.provenance.document_id",
+    "cell_suspension.genus_species.ontology",
+    "cell_suspension.genus_species.ontology_label",
+    "donor_organism.provenance.document_id",
+    "donor_organism.human_specific.ethnicity.ontology",
+    "donor_organism.human_specific.ethnicity.ontology_label",
+    "donor_organism.diseases.ontology",
+    "donor_organism.diseases.ontology_label",
```

Comment:
The field used to populate the browser for 'Disease Status' is specimen_from_organism.diseases.ontology_label. This is because there can be 'normal' specimen collected from donors with other non-related diseases. Would it be worthwhile to also include specimen disease ontology information?

Reply:
Ah, good catch! We do actually store both in our index and they're available to filter on, but it appears we haven't updated `GET /filters` to return this.

Reply:
[Ticket](https://app.zenhub.com/workspaces/dcp-5ac7bcf9465cb172b77760d9/issues/humancellatlas/matrix-service/430). Resolving